clients/cli/src/main.rs

```diff
@@ -1,265 +1,245 @@
-        /// The lockbook location of a file within lockbook
-        path: Option<String>,
+fn resolve_target_to_id(core: &Core, t: &str) -> Result<lb::Uuid, CliError> {
+    if let Ok(id) = lb::Uuid::parse_str(t) {
+        return Ok(id);
+    }
+    match core.get_by_path(t) {
+        Ok(f) => Ok(f.id),
+        Err(err) => Err((err, t).into()),
+    }
+}
 
-        /// The lockbook location of a file within lockbook
-        #[structopt(short, long)]
-        id: Option<Uuid>,
+fn delete(core: &Core, target: &str, force: bool) -> Result<(), CliError> {
+    let f = resolve_target_to_file(core, target)?;
+
+    if !force {
+        let mut phrase = format!("delete '{}'", target);
+        if f.is_folder() {
+            let count = core
+                .get_and_get_children_recursively(f.id)
+                .map_err(|err| (err, f.id))?
+                .len();
+            phrase = format!("{phrase} and its {count} children")
+        }
+
+        let answer: String = input(format!("are you sure you want to {phrase}? [y/n]: "))?;
+        if answer != "y" && answer != "Y" {
+            println!("aborted.");
+            return Ok(());
+        }
+    }
 
-        /// New name
-        name: Option<String>,
-    },
+    core.delete_file(f.id).map_err(|err| (err, f.id).into())
+}
 
-    /// Manage shared documents
-    Share(share::Share),
+fn move_file(core: &Core, src: &str, dest: &str) -> Result<(), CliError> {
+    let src_id = resolve_target_to_id(core, src)?;
+    let dest_id = resolve_target_to_id(core, dest)?;
+    core.move_file(src_id, dest_id)
+        .map_err(|err| (err, src_id, dest_id).into())
+}
 
-    /// What operations a sync would perform
-    Status,
+fn print(core: &Core, target: &str) -> Result<(), CliError> {
+    let id = resolve_target_to_id(core, target)?;
+    let content = core.read_document(id).map_err(|err| (err, id))?;
+    print!("{}", String::from_utf8_lossy(&content));
+    io::stdout().flush()?;
+    Ok(())
+}
 
-    /// Get updates, push changes
-    Sync,
+fn rename(core: &Core, target: &str, new_name: &str) -> Result<(), CliError> {
+    let id = resolve_target_to_id(core, target)?;
+    core.rename_file(id, new_name)
+        .map_err(|err| (err, id).into())
+}
 
-    /// Subcommands that aid in extending Lockbook
-    Debug(debug::Debug),
+fn sync(core: &Core) -> Result<(), CliError> {
+    println!("syncing...");
+    core.sync(Some(Box::new(|sp: lb::SyncProgress| {
+        use lb::ClientWorkUnit::*;
+        match sp.current_work_unit {
+            PullMetadata => println!("pulling file tree updates"),
+            PushMetadata => println!("pushing file tree updates"),
+            PullDocument(name) => println!("pulling: {}", name),
+            PushDocument(name) => println!("pushing: {}", name),
+        };
+    })))?;
+    Ok(())
 }
 
-fn exit_with(err: CliError) -> ! {
-    err.print();
-    std::process::exit(err.code as i32)
+fn create(core: &Core, path: &str) -> Result<(), CliError> {
```

Comment:
feels weird that some of the fns are here, no?

Reply:
Not weirder than splitting them up into many small files imo. Your call